neqo-transport/src/send_stream.rs

```diff
@@ -992,6 +1029,7 @@ impl SendStream {
                 let sent = send_buf.send(buf);
                 fc.consume(sent);
                 conn_fc.borrow_mut().consume(sent);
+                self.bytes_written += sent as u64;
```

Comment:
You should be able to track this on a send buffer as `self.retired + self.send_buf.len()` (with appropriate casting) without the additional variable.

If you need to retain this, then keeping it on `SendState`, like you have done for the retired value, should be fine.  It's a little more work to get, but less risk of something getting desynchronized.